lib/xebow_web/live/matrix_live.ex

```diff
@@ -37,7 +37,8 @@ defmodule XebowWeb.MatrixLive do
 
   @impl Phoenix.LiveView
   def handle_info({:render, frame}, socket) do
-    {:noreply, assign(socket, leds: make_view_leds(frame))}
+    colors = for led <- make_view_leds(frame), into: %{}, do: {led.id, led.color}
```

Comment:
:information_source: (optional)

This `for` statement is crafty, but it's also a lot of information packed into one line. An alternative is to split the code up into a pipeline to spread the steps over multiple lines.

Here is the literal translation:

```suggestion
    colors =
      frame
      |> make_view_leds()
      |> Enum.map(fn led -> {led.id, led.color} end)
      |> Enum.into(%{})
```

Or there's this one that omits `into` by using `reduce`:

```suggestion
    colors =
      frame
      |> make_view_leds()
      |> Enum.reduce(%{}, fn led, acc -> Map.put(acc, led.id, led.color) end)
```

Reply:
From an AST point of view, the list comprehension is tighter, but we can go with the easier to read one.